dask_jobqueue/tests/test_job.py

```diff
@@ -1,61 +1,24 @@
 import asyncio
 from time import time
 
-from dask_jobqueue import (
-    PBSCluster,
-    SGECluster,
-    SLURMCluster,
-    LSFCluster,
-    HTCondorCluster,
-    MoabCluster,
-    OARCluster,
-)
-from dask_jobqueue.local import LocalJob, LocalCluster
+from dask_jobqueue.local import LocalCluster
 from dask_jobqueue.pbs import PBSJob
-from dask_jobqueue.sge import SGEJob
-from dask_jobqueue.slurm import SLURMJob
-from dask_jobqueue.lsf import LSFJob
-from dask_jobqueue.moab import MoabJob
-from dask_jobqueue.htcondor import HTCondorJob
-from dask_jobqueue.oar import OARJob
-
 from dask_jobqueue.core import JobQueueCluster
 from dask.distributed import Scheduler, Client
 from distributed.core import Status
 
 import pytest
 
 
-def test_basic():
-    job = PBSJob(scheduler="127.0.0.1:12345", cores=1, memory="1 GB")
+def test_basic(Cluster):
+    job_cls = Cluster.job_cls
+    job = job_cls(scheduler="127.0.0.1:12345", cores=1, memory="1 GB")
     assert "127.0.0.1:12345" in job.job_script()
 
 
-job_protected = [
-    pytest.param(SGEJob, marks=[pytest.mark.env("sge")]),
-    pytest.param(PBSJob, marks=[pytest.mark.env("pbs")]),
-    pytest.param(SLURMJob, marks=[pytest.mark.env("slurm")]),
-    pytest.param(LSFJob, marks=[pytest.mark.env("lsf")]),
-    LocalJob,
-]
-
-
-all_jobs = [SGEJob, PBSJob, SLURMJob, LSFJob, HTCondorJob, MoabJob, OARJob]
-all_clusters = [
-    SGECluster,
-    PBSCluster,
-    SLURMCluster,
-    LSFCluster,
-    HTCondorCluster,
-    MoabCluster,
-    OARCluster,
-    HTCondorCluster,
-]
-
-
-@pytest.mark.parametrize("job_cls", job_protected)
 @pytest.mark.asyncio
-async def test_job(job_cls):
+async def test_job(EnvSpecificCluster):
+    job_cls = EnvSpecificCluster.job_cls
```

Comment:
The other changes are similar: The logic `pytest.mark.env` logic is moved into the EnvSpecificClusterFixture